packages/router/__tests__/router-test.ts

```diff
@@ -10601,19 +10635,78 @@ describe("a router", () => {
         );
       });
 
+      it("should handle not found routes with a 404 Response", async () => {
+        /* eslint-disable jest/no-conditional-expect */
+        let { queryRoute } = createStaticHandler([
+          {
+            id: "root",
+            path: "/",
+          },
+        ]);
+
+        try {
+          await queryRoute(createRequest("/junk"));
+          expect(false).toBe(true);
+        } catch (data) {
+          expect(data instanceof Response).toBe(true);
+          expect(data.status).toBe(404);
+          expect(data.statusText).toBe("Not Found");
+          expect(data.headers.get("X-Remix-Router-Error")).toBe("yes");
```

Comment:
Router-generated errors (404 and 405) will now contain a custom header we can leverage in remix as needed

Reply:
This should cover our use-case :)